module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -1439,20 +1439,21 @@ public function patronLogin($username, $password)
                 }
             }
         }
-
-        return [
-            'id' => $profile->id,
-            'username' => $username,
-            'cat_username' => $username,
-            'cat_password' => $password,
-            'firstname' => $profile->personal->firstName ?? null,
-            'lastname' => $profile->personal->lastName ?? null,
-            'email' => $profile->personal->email ?? null,
-            'addressTypeIds' => array_map(
-                fn ($address) => $address->addressTypeId,
-                $profile->personal->addresses ?? []
-            ),
-        ];
+        $patron = $this->createPatronArray(
+            id: $profile->id,
+            cat_username: $username,
+            cat_password: $password,
+            firstname: $profile->personal->firstName,
+            lastname: $profile->personal->lastName,
```

Comment:
Is there a reason you removed the defaulting to null here? I think it may be needed to prevent notices if the response lacks a personal section.

```suggestion
            firstname: $profile->personal->firstName ?? null,
            lastname: $profile->personal->lastName ?? null,
```